src/main/java/com/microsoft/azure/util/AzureCredentials.java

```diff
@@ -530,7 +530,14 @@ public static TokenCredential getSystemCredentialById(String credentialID) {
                 CredentialsMatchers.withId(credentialID));
 
         if (!azureImdsCredentials.isEmpty()) {
-            return new ManagedIdentityCredentialBuilder().build();
+            AzureImdsCredentials azureIdmsCredential = azureImdsCredentials.get(0);
+            ManagedIdentityCredentialBuilder credentialBuilder = new ManagedIdentityCredentialBuilder();
+
+            if (azureIdmsCredential.isUserAssigned()) {
```

Comment:
this should check if client ID is non empty

Reply:
I changed it to check that the variable is not null (because I used fixEmpty in the setter).
Should I check for null+blank or just checking for null is enough ?